flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/runtime/aggregate/MaxAggregate.scala

```diff
@@ -48,8 +57,16 @@ abstract class MaxAggregate[T](implicit ord: Ordering[T]) extends Aggregate[T] {
   override def merge(intermediate: Row, buffer: Row): Unit = {
```

Comment:
I think we can make this method a bit more efficient:
```
val partialValue = ...
if (partialValue != null) {
  val bufferValue = ...
  if (bufferValue != null) {
    val max = ...
    buffer.setField(maxIndex, max)
  } else {
    buffer.setField(maxIndex, partialValue)
  }
}
```

Same applies for `MinAggregate` and `SumAggregate`.